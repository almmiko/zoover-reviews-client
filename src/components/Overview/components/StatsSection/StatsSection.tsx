import React from 'react';
import {
  Wrapper,
  Title,
  Container,
  Section,
  Row,
  StatTitle,
  StatValue,
  Header
} from './elements';
import { STATS_MAPPING } from '../../../../stores/reviewsStore';

type Props = {
  loaded: boolean,
  title: string,
  stats: {[ key: string ]: any} | undefined,
}

class StatsSection extends React.Component<Props> {
  render() {
    const { title, stats = {}, loaded } = this.props;

    return (
      <Wrapper>
        <Header>
          <Title>{title}</Title>
        </Header>
        <Container>
          <Section>
          {loaded ? Object.keys(stats).map((stat: string) => {
            const statValue = stats[stat];
            if (!statValue) return null;

            return (
              <Row key={statValue}>
                <StatTitle>{STATS_MAPPING[stat] || stat}</StatTitle>
                <StatValue>{statValue}</StatValue>
              </Row>
            )
          }): 'Loading...'}
          </Section>
        </Container>
      </Wrapper>
    );
  }
}

export default StatsSection;
