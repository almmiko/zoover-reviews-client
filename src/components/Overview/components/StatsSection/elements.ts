import styled from 'styled-components';

export const Wrapper = styled.div`
  position: relative;
  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, .14);
  background-color: #fff;
  padding: 15px;  
`;

export const Header = styled.div`
  top: 0;
  left: 50%;
  width: 95%;
  position: absolute;
  transform: translate(-50%, -50%);
  padding: 20px;
  border-radius: 4px;
  background: #fd9810;
  box-shadow: 0 12px 20px -10px rgba(255,165,115,0.28), 
              0 4px 20px 0 rgba(0, 0, 0, .12),
              0 7px 8px -5px rgba(255,165,115,0.2);
`;

export const Title = styled.div`
  font-family: 'Open Sans', sans-serif;
  font-weight: bold;
  color: #fff;
  font-size: 16px;
  text-transform: uppercase;
`;

export const Container = styled.div`
  display: flex;
  width: 100%;
  margin-top: 30px;
`;

export const Row = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
  padding: 5px 0;
  border-bottom: 1px solid #d1d1d2;
  
  &:last-child {
    border-bottom: none;
  }
`;

export const Section = styled.div`
  margin-right: 20px;
  width: 100%;
  
  &:last-child {
    margin-right: 0;
  }
`;

export const StatValue = styled.div`
  font-size: 15px;
  color: #757575;
`;

export const StatTitle = styled.div`
  font-size: 15px;
  color: #9e9e9e;
  text-transform: capitalize;
`;
