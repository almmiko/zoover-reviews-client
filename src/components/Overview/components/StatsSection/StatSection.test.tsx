import { shallow } from 'enzyme';
import StatsSection from './StatsSection';
import React from 'react';

it('renders correctly', () => {
  const tree = shallow(<StatsSection
    apiError={false}
    loaded={true}
    stats={{ test: 'test' }}
    title={'test'} />);
  expect(tree).toMatchSnapshot();
});
