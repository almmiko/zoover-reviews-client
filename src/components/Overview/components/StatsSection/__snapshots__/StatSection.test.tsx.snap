// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      test
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <styled.div
        key="test"
      >
        <styled.div>
          test
        </styled.div>
        <styled.div>
          test
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
